web/packages/api/src/toEthereum_v2.ts

```diff
@@ -347,38 +374,84 @@ export async function getDeliveryFee(
         snowbridgeDeliveryFeeDOT = BigInt(leFee.toString())
     }
 
-    const xcm = buildResultXcmAssetHubERC20TransferFromParachain(
-        assetHub.registry,
-        registry.ethChainId,
-        "0x0000000000000000000000000000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000000000000000000000000000",
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        parachain,
-        340282366920938463463374607431768211455n
-    )
+    const { tokenErcMetadata, sourceParachain, ahAssetMetadata, sourceAssetMetadata } =
+        resolveInputs(registry, tokenAddress, parachain)
 
-    let assetHubExecutionFeeDOT = 0n
-    let returnToSenderExecutionFeeDOT = 0n
-    let returnToSenderDeliveryFeeDOT = 0n
-    const bridgeHubDeliveryFeeDOT = await calculateDeliveryFee(
-        assetHub,
-        registry.bridgeHubParaId,
-        xcm
-    )
-    if (parachain !== registry.assetHubParaId) {
-        const returnToSenderXcm = buildParachainERC20ReceivedXcmOnDestination(
+    let xcm: any
+
+    if (sourceAssetMetadata.location) {
+        xcm = buildResultXcmAssetHubPNATransferFromParachain(
             assetHub.registry,
             registry.ethChainId,
+            sourceAssetMetadata,
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
             "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
             340282366920938463463374607431768211455n,
             340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n,
+            parachain,
+            340282366920938463463374607431768211455n
+        )
+    } else {
+        xcm = buildResultXcmAssetHubERC20TransferFromParachain(
+            assetHub.registry,
+            registry.ethChainId,
             "0x0000000000000000000000000000000000000000000000000000000000000000",
-            "0x0000000000000000000000000000000000000000000000000000000000000000"
+            "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n,
+            parachain,
+            340282366920938463463374607431768211455n
+        )
+    }
+    console.log("xcm on asset hub for pna transfer from 3rd parachain:", xcm.toHuman())
```

Comment:
this will show in the debug console in the browser. So maybe remove logs like this.

Reply:
https://github.com/Snowfork/snowbridge/pull/1419/commits/d46b66e5ec02da79143dfb32d4a3080b94baf0f9